### ISSUE TITLE
Android: Disabled States does not announce or disable function for some components

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30840]
## Description
Does not announce "disabled" on the following components:

- [x] Button #30934
- [x] Image  #30935
- [x] ImageBackground #30936
- [ ] Text #30937
- [ ] TextInput  #30938
- [ ] Slider #30939
- [x] Picker #30940
- [ ] ProgressBarAndroid #30941

Doesn't disable click functionality of the following components:
- [x]  Button #30943
- [ ]  Switch #30944
- [x]  Slider  #30945
- [x]  Picker #30946 (Picker's own "enabled" property works correctly here)
- [x]  Text  #30947
- [ ]  TextInput #30948
- [x]  Pressable #30949
- [x]  TouchableHighlight  #30950
- [x]  TouchableOpacity #30951
- [x]  TouchableNativeFeedback #30952
- [x]  TouchableWithoutFeedback  #30953

This behavior is usually due to there being both a "disabled" prop on the component itself, that does not necessarily sync with the accessibilityState's "disabled" field. When "disabled" is set, we should just automatically set accessibilityState: disabled as well so that they stay in sync. (You can see an example of this fix in the linked Pressable issue).

When testing a component, be sure to test both the accessibilityState and the components own disabled prop if one exists, and make sure all combinations of them work as expected. [This commit](https://github.com/facebook/react-native/commit/f69e096bb4df67474351786f674b1bb1e42d3363) has a great example of snapshot tests added to ensure the component behaves as expected.

Issue #30246 is similar to this issue and should be taken into account when creating a solution.

## React Native version:
v0.63

## Expected Results

When a component is "disabled", screen readers should announce this as part of their focus announcement. Clicks to these elements should also have no effect.

If a component has both a "disabled" prop and allows "accessibilityState: disabled", the "disabled" prop should be the authority on the state of the component (ignoring accessibilityState: disabled).

## Snack
https://snack.expo.io/d1Ikg7nqo

## Android details
To disable an element for Talkback, the `AccessibilityNodeInfo`'s `enabled` property should be set to false and the `clickable`, `longClickable`, or `focusable` attribute should be true (one of these is likely already true if the element was clickable, it's important that it remains that way).  

If an `AccessibilityAction` of `AccessibilityNodeInfo.ACTION_CLICK`, `AccessibilityNodeInfo.ACTION_LONG_CLICK`, or `AccessibilityAction.ACTION_CONTEXT_CLICK` is set on the `AccessibilityNodeInfo`, it should be removed.

All of the above can be achieved automatically by simply setting the `enabled` property of the View itself to false, but this means that the components own "disabled" prop (if one exists, such as with <Button>), and the accessibilityState's "disabled" field, must remain in sync.

**Comment by lunaleaps** [comment]
@blavalla Is the expectation that `accessibilityState: disabled` and `disabled` prop to function the same way? What is the purpose of having both options then? Should we consolidate to one way?

**Comment by blavalla** [comment]
@lunaleaps, I would expect both properties to have the same experience. As for why they are separate, it's likely because the "disabled" prop also applies some sort of visual indicator of its state (dimming the element, etc.) which is what it is often used for. The `accessibilityState: disabled` *only* applies to the accessibility system, so won't do anything visually to the element itself. The `accessibilityState: disabled` prop is also part of a set of props that are on many different components ([here](https://github.com/facebook/react-native/blob/93377ff5089d981e56094329b6f6893b3124be12/Libraries/Components/View/ViewPropTypes.js#L379)), and `disabled` seems to be something implemented on a component-by-component basis.

When `disabled` is set we should probably just automatically set `accessibilityState: disabled` as well, or alternatively in the mapping of RN props to the native Views, we should consider this prop in the same way we consider `accessibilityState: disabled`. I can't think of a valid reason for these two properties to ever be out of sync, so we should probably make sure they don't get that way.

**Comment by fabriziobertoglio1987** [comment]
@blavalla 

I published PRs https://github.com/facebook/react-native/pull/33076 https://github.com/facebook/react-native/pull/33070 https://github.com/callstack/react-native-slider/pull/354 to fix this issues.
 
The PRs are in DRAFT. I will complete the PRs in a couple of days. 
Thanks :pray:
